fs-api/src/System/FS/IO.hs

```diff
@@ -32,7 +31,7 @@ import qualified System.FS.IO.Internal.Handle as H
 -- We store the path the handle points to for better error messages
 type HandleIO = F.FHandle
 
-ioHasFS :: MonadIO m => MountPoint -> HasFS m HandleIO
+ioHasFS :: (MonadIO m, PrimState IO ~ PrimState m) => MountPoint -> HasFS m HandleIO
 ioHasFS mount = HasFS {
```

Comment:
I wonder if `ioHasRealFS` would be a more fitting name, since both `ioHasFS` and `simHasFS` can be used in `IO`.

Reply:
Or just `realFS`, io can be inferred from the type.

Reply:
True. I might change it in a separate PR